server.js

```diff
@@ -336,9 +337,14 @@ Server.prototype._onWebSocketRequest = function (socket, opts, params) {
     }
     if (self.destroyed) return
 
-    if (socket.infoHashes.indexOf(params.info_hash) === -1) {
-      socket.infoHashes.push(params.info_hash)
-    }
+    var hashes
+    if (typeof params.info_hash === 'string') hashes = [ params.info_hash ]
+    else hashes = params.info_hash
```

Comment:
If `params.info_hash` is not an Array, then the server will crash on the next line. Better to check with `Array.isArray`.